shared-module/displayio/Display.c

```diff
@@ -141,11 +151,13 @@ void common_hal_displayio_display_construct(displayio_display_obj_t* self,
         } else {
             self->backlight_pwm.base.type = &pulseio_pwmout_type;
             common_hal_pulseio_pwmout_never_reset(&self->backlight_pwm);
-            if (!self->auto_brightness) {
-                common_hal_displayio_display_set_brightness(self, brightness);
-            }
         }
     }
+    if (!self->auto_brightness && (self->backlight_inout.base.type != &mp_type_NoneType || brightness_command <= 0xff)) {
+        common_hal_displayio_display_set_brightness(self, brightness);
+    } else {
+        self->current_brightness = -1.0;
```

Comment:
If brightness is -1.0, would that mess up the brightness calculation below where it's multiplied by 0xff? Or is this a illegal value that should get checked for somewhere?

Reply:
Added a check to the setter in shared-bindings. This value isn't actually used to set the brightness. It is only returned to indicate brightness isn't controllable.